src/modules/Dropdown/Dropdown.js

```diff
@@ -100,6 +100,21 @@ export default class Dropdown extends Component {
     /** Name of the hidden input which holds the value. */
     name: PropTypes.string,
 
+    /**
+     * Allow user additions to the list of options (boolean).
+     * Requires the use of `selection`, `options` and `search`.
+     */
+    allowAdditions: customPropTypes.every([
+      customPropTypes.demand(['options', 'selection', 'search']),
+      PropTypes.bool,
+    ]),
+
+    /** Called with the new value added by the user. Use this to update the options list. */
+    onAddition: PropTypes.func,
```

Comment:
Looking back on this now, this prop probably makes more sense as `onAddItem`.  Mind updating?

_Side note, I plan on updating the `options` prop to `items` so it follows SUI closer.  Then the, `Select` component will use `options`._